### PR TITLE
add edgecore image build script

### PR DESCRIPTION
**Pull request by TianTianBigWang**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: build/docker/edgeimg-make.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  echo "docker check failed"
+  exit 1
+fi
+
+go mod vendor
+
+docker buildx build --build-arg ARCH=amd64 --platform amd64 -t "$REPOSITORY":"$RELEASE_VERSION"-amd64 -f "$WORK_DIR/edgecore.dockerfile" -o type=docker .
+docker buildx build --build-arg ARCH=arm64 --platform arm64 -t "$REPOSITORY":"$RELEASE_VERSION"-arm64 -f "$WORK_DIR/edgecore.dockerfile" -o type=docker .
+docker buildx build --build-arg ARCH=arm --platform arm -t "$REPOSITORY":"$RELEASE_VERSION"-arm -f "$WORK_DIR/edgecore.dockerfile" -o type=docker .
+
```
Comment (chendave) [inline build/docker/edgeimg-make.sh:20]: can you pls define a platform list and loop against that list?

File: build/docker/edgeimg-make.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+
+REPOSITORY=${REPOSITORY:-"kubeedge/edgecore"}
+RELEASE_VERSION=${RELEASE_VERSION:-"v1.9.1"}
```
Comment (chendave) [inline build/docker/edgeimg-make.sh:6]: seems that we need update this version after each release?

File: build/docker/edgecore.dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    elif [ "${ARCH}" = "arm64" ]; then make crossbuild ARM_VERSION=GOARM8 WHAT=edgecore && cp _output/local/bin/edgecore bin/edgecore ; \
+    elif [ "${ARCH}" = "arm" ]; then make crossbuild ARM_VERSION=GOARM7 WHAT=edgecore && cp _output/local/bin/edgecore bin/edgecore;  \
+    fi
+
+FROM ubuntu:18.04
+COPY --from=builder /work/_output/local/bin/edgecore /usr/local/bin/edgecore
+
+WORKDIR /etc/kubeedge
+# Custom image can add more content here.
+# e.g. config
```
Comment (chendave) [inline build/docker/edgecore.dockerfile:16]: add a line break here?

File: build/docker/edgecore.dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
+FROM kubeedge/build-tools:latest as builder
+ARG ARCH=amd64
+WORKDIR /work
+ADD . .
+RUN mkdir -p bin &&\
+    if [ "${ARCH}" = "amd64" ]; then make WHAT=edgecore && cp _output/local/bin/edgecore bin/edgecore ; \
+    elif [ "${ARCH}" = "arm64" ]; then make crossbuild ARM_VERSION=GOARM8 WHAT=edgecore && cp _output/local/bin/edgecore bin/edgecore ; \
+    elif [ "${ARCH}" = "arm" ]; then make crossbuild ARM_VERSION=GOARM7 WHAT=edgecore && cp _output/local/bin/edgecore bin/edgecore;  \
```
Comment (fisherxu) [inline build/docker/edgecore.dockerfile:8]: Now the build-tools base has supported multi-CPU architecture, so no need to do the crossbuild here :)

File: build/docker/edgeimg-make.sh (hunk 2 above)
Comment (fisherxu) [reply build/docker/edgeimg-make.sh:6]: Should got from git tag :)

File: build/docker/edgeimg-make.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+pushTag="$1"
+WORK_DIR=$(cd "$(dirname "$0")";pwd)
+
+if ! [ "$(docker version)" ]; then
+  echo "docker check failed"
+  exit 1
+fi
+
+go mod vendor
```
Comment (fisherxu) [inline build/docker/edgeimg-make.sh:15]: KubeEdge already has the vendor

File: build/docker/edgecore.dockerfile (hunk 4 above)
Comment (fisherxu) [reply build/docker/edgecore.dockerfile:9]: Any comments about this?

**Review by fisherxu — COMMENTED** [review]
keadm should also included in this image? If so we need to rename the image, any ideas? @zc2638 @zhu733756 @TianTianBigWang

File: build/docker/edgeimg-make.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+
+REPOSITORY=${REPOSITORY:-"kubeedge/edgecore"}
+RELEASE_VERSION=$(git describe --tags `git rev-list --tags --max-count=1`)
```
Comment (fisherxu) [inline build/docker/edgeimg-make.sh:6]:
```suggestion
RELEASE_VERSION=$(git describe --tags)
```
Use `git describe --tags` should be more accurate, which reflects the latest commit situation.

File: build/docker/edgeimg-make.sh [hunk 7]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+
+REPOSITORY=${REPOSITORY:-"kubeedge/edgecore"}
```
Comment (fisherxu) [inline build/docker/edgeimg-make.sh:5]: Let's named the repo to `kubeedge/installation-package` :)  And place the keadm in `/usr/local/bin/`.

**Comment by fisherxu** [comment]
The dockerfile and script can move to a subdir `installation-package`

**Comment by TianTianBigWang** [comment]
all done. PTAL

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**